### DISCUSSION TITLE
SMTP - TLS (suggestion)

### DISCUSSION THREAD

**Discussion by lorol** [timschofield/webERP#165] [Feature Requests]
Hi, 
I managed to ugly-patch htmlMimeMail.php and smtp.php to be able to test Gmail / Hotmail free SMTP services. 

Of course, it is not very serious for any manufacturing company to use them, neither to use an ERP on free hosting :) but anyway, someone else may want to try or use SMTP w/TLS e-mailing 
 
Notes - issues faced:
- STARTTLS on port 587 is easy to be added add to the existing smtp.php
- However, these free SMTP (Gmail Hotmail) services are restrictive, so the 'From' should always be the same (as the SMTP login username) 
- Also helo string in webERP may need longer size, better to use again a fixed one within the code - entire SMTP setup page needs a review, BTW
- There is a problem ( Shared hosting w/ PHP 7.4 environment ) htmlMimeMail.php to call statically this, like smtp::  and  Mail_RFC822:: functions so I needed to place some workaround code, too. 

Maybe a better replacement can be done w/ PHPmailer plus the necessary addition SMTP configuration fields. 
Please consider.

**Comment by lorol** [comment]
Nevermind - jus saw the other project :)

**Reply by dalers** [reply]
Please, what other project?

**Comment by dalers** [comment]
SMTP mail works now with Gmail / Hotmail using TLS per https://github.com/timschofield/webERP/discussions/396
